lib/galaxy/security/validate_user_input.py

```diff
@@ -45,7 +45,7 @@ def validate_publicname( trans, publicname, user=None ):
         if len( publicname ) < 3:
             return "Public name must be at least 3 characters in length"
     else:
-        if len( publicname ) < 4:
+        if len( publicname ) < 3:
             return "Public name must be at least 4 characters in length"
```

Comment:
Good point. I would not rush into lowering the requirements of 4 char min as others might chime in with different insights.